src/status_im/contexts/wallet/send/select_address/view.cljs

```diff
@@ -23,47 +23,52 @@
-        :on-blur               #(reset! input-focused? false)
-        :on-scan               (fn []
-                                 (rn/dismiss-keyboard!)
-                                 (rf/dispatch [:wallet/clean-scanned-address])
-                                 (rf/dispatch [:open-modal :screen/wallet.scan-address]))
-        :ens-regex             constants/regx-ens
-        :scanned-value         (or (when recipient-plain-address? send-address) scanned-address)
-        :address-regex         constants/regx-multichain-address
-        :on-detect-address     #(when (or (= current-screen-id :screen/wallet.select-address)
-                                          (= current-screen-id :screen/wallet.scan-address))
-                                  ; ^ this check is to prevent effect being triggered when screen is
-                                  ; loaded but not being shown to the user (deep in the navigation
-                                  ; stack) and avoid undesired behaviors
-                                  (debounce/debounce-and-dispatch
-                                   [:wallet/validate-address %]
-                                   300))
-        :on-detect-ens         (fn [text cb]
-                                 (when (or (= current-screen-id :screen/wallet.select-address)
-                                           (= current-screen-id :screen/wallet.scan-address))
-                                   ; ^ this check is to prevent effect being triggered when screen
-                                   ; is loaded but not being shown to the user (deep in the
-                                   ; navigation stack) and avoid undesired behaviors
-                                   (debounce/debounce-and-dispatch
-                                    [:wallet/find-ens text contacts chain-id cb]
-                                    300)))
-        :on-change-text        (fn [text]
-                                 (when (empty? text)
-                                   (rf/dispatch [:wallet/clean-local-suggestions]))
-                                 (reset! input-value text))
-        :valid-ens-or-address? valid-ens-or-address?}])))
+  (let [set-scanned-value-fn (atom nil)]
+    (fn []
+      (let [view-id                  (rf/sub [:view-id])
+            scanned-address          (rf/sub [:wallet/scanned-address])
+            send-address             (rf/sub [:wallet/wallet-send-to-address])
+            recipient                (rf/sub [:wallet/wallet-send-recipient])
+            recipient-plain-address? (= send-address recipient)
+            valid-ens-or-address?    (rf/sub [:wallet/valid-ens-or-address?])
+            chain-id                 (rf/sub [:chain-id])
+            contacts                 (rf/sub [:contacts/active])]
+        (when (and @set-scanned-value-fn scanned-address)
+          (@set-scanned-value-fn scanned-address)
+          (reset! set-scanned-value-fn nil))
```

Comment:
Maybe it is more natural having a `use-effect` here?

Reply:
please check proper fix here https://github.com/status-im/status-mobile/pull/19341